ddtrace/tracer.py

```diff
@@ -307,7 +320,13 @@ def configure(
             report_metrics=config.health_metrics_enabled,
         )
         self.writer.dogstatsd = get_dogstatsd_client(self._dogstatsd_url)
-        self.processor = TraceProcessor(filters=self._filters)  # type: ignore[call-arg]
+        self.deregister_on_start_span(self.processor.on_span_start)
+        self.processor = SpanProcessor(  # type: ignore[call-arg]
+            filters=self._filters,
+            partial_flush_enabled=get_partial_flush_enabled(),
+            partial_flush_min_spans=get_partial_flush_min_spans(),
+        )
+        self.on_start_span(self.processor.on_span_start)
```

Comment:
Given that the tracer knows about the processor, I wonder if it wouldn't make more sense to call `processor.on_span_start` directly from `start_span` rather than  use the `on_start_span` callback

Reply:
yeah fair point, it's clunky to have to register and deregister hooks when we do have a processor :+1: